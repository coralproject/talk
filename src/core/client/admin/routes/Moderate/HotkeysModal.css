.root {
  min-width: 475px;
}

.hotKeyContainer {
  width: 65px;
}

.hotKey {
  composes: bodyCopy from "coral-ui/shared/typography.css";
  background-color: #f2f2f2;
  border: 1px solid #bbbebf;
  border-radius: 3px;
  padding: var(--spacing-1);
  display: inline-block;
  line-height: 1;
}