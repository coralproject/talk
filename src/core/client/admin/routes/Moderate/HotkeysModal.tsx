import { Localized } from "@fluent/react/compat";
import React, { FunctionComponent } from "react";

import {
  Card,
  CardCloseButton,
  Flex,
  HorizontalGutter,
  Modal,
} from "coral-ui/components/v2";

import styles from "./HotkeysModal.css";

interface Props {
  open: boolean;
  onClose: () => void;
}

const HotkeysModal: FunctionComponent<Props> = ({ open, onClose }) => {
  return (
    <Modal open={open} onClose={onClose} aria-labelledby="banModal-title">
      {({ firstFocusableRef }) => (
        <Card className={styles.root}>
          <CardCloseButton onClick={onClose} ref={firstFocusableRef} />
          <HorizontalGutter spacing={3}>
            <Localized id="hotkeysModal-title">
              <h1 className={styles.title}>Keyboard shortcuts</h1>
            </Localized>
            <Flex itemGutter>
              <HorizontalGutter spacing={2}>
                <Localized id="hotkeysModal-navigation-shortcuts">
                  <div className={styles.subTitle}>Navigation shortcuts</div>
                </Localized>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>j</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-next">
                    <div className={styles.hotKeyAction}>Next comment</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>k</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-prev">
                    <div className={styles.hotKeyAction}>Previous comment</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>ctrl+f</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-search">
                    <div className={styles.hotKeyAction}>Open search</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>1..4</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-jump">
                    <div className={styles.hotKeyAction}>
                      Jump to specific queue
                    </div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>t</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-switch">
                    <div className={styles.hotKeyAction}>Switch queues</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>z</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-zen">
                    <div className={styles.hotKeyAction}>
                      Toggle single-comment view
                    </div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>?</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-toggle">
                    <div className={styles.hotKeyAction}>
                      Toggle shortcuts help
                    </div>
                  </Localized>
                </Flex>
              </HorizontalGutter>
              <HorizontalGutter spacing={2}>
                <Localized id="hotkeysModal-moderation-decisions">
                  <div className={styles.subTitle}>Moderation decisions</div>
                </Localized>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>f</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-approve">
                    <div className={styles.hotKeyAction}>Approve</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>d</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-reject">
                    <div className={styles.hotKeyAction}>Reject</div>
                  </Localized>
                </Flex>
                <Flex alignItems="center">
                  <div className={styles.hotKeyContainer}>
                    <div className={styles.hotKey}>b</div>
                  </div>
                  <Localized id="hotkeysModal-shortcuts-ban">
                    <div className={styles.hotKeyAction}>
                      Ban comment author
                    </div>
                  </Localized>
                </Flex>
              </HorizontalGutter>
            </Flex>
          </HorizontalGutter>
        </Card>
      )}
    </Modal>
  );
};

export default HotkeysModal;
