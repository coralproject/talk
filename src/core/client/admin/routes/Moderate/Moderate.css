$moderate-background: var(--palette-grey-100);

.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background-color: $moderate-background;
}

.main {
  margin: var(--spacing-5) 0 var(--spacing-7) 0;
  display: flex;
  justify-content: center;
}

.subBar {
  background-color: $moderate-background;
}
