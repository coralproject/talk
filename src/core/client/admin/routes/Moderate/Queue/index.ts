export {
  PendingQueueRoute,
  UnmoderatedQueueRoute,
  ReportedQueueRoute,
} from "./QueueRoute";
export { default as RejectedQueueRoute } from "./RejectedQueueRoute";
export { routeConfig as RejectedQueueRouteConfig } from "./RejectedQueueRoute";
export { default as ApprovedQueueRoute } from "./ApprovedQueueRoute";
export { routeConfig as ApprovedQueueRouteConfig } from "./ApprovedQueueRoute";
export { default as LoadingQueue } from "./LoadingQueue";
export { default as ModerationQueue } from "./ModerationQueueQuery";
export { default as Queue } from "./Queue";
export { routeConfig as ForReviewQueueRouteConfig } from "./ForReviewQueueRoute/ForReviewQueueRoute";
