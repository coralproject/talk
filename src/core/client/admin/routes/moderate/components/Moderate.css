.background {
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background-color: var(--palette-background-light);
}

.main {
  margin: calc(2 * var(--spacing-unit)) 0 calc(4 * var(--spacing-unit)) 0;
  display: flex;
  justify-content: center;
}

.modQueue {
  width: calc(75 * var(--spacing-unit));
}

.topBar {
  margin-bottom: var(--spacing-unit);
}

.username {
  margin-right: var(--spacing-unit);
}

.footer {
  margin-top: calc(2 * var(--spacing-unit));
}

.content {
  min-height: calc(7.5 * var(--spacing-unit));
}

.aside {
  padding-top: 25px;
  flex-shrink: 0;
  flex-grow: 0;
  box-sizing: border-box;
}

.reject {
  border: 1px solid var(--palette-error-main);
  box-sizing: border-box;
  border-radius: 2px;
  width: 65px;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: var(--palette-error-main);
}

.approve {
  border: 1px solid var(--palette-success-main);
  box-sizing: border-box;
  border-radius: 2px;
  width: 65px;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: var(--palette-success-main);
}

.icon {
  font-weight: var(--font-weight-bold);
}

.decision {
  font-size: calc(14rem / var(--rem-base));
  font-weight: var(--font-weight-medium);
  font-family: var(--font-family-sans-serif);
  line-height: calc(16em / 14);
  letter-spacing: calc(0.2em / 14);
  color: var(--palette-text-primary);
  text-transform: uppercase;
}

.separator {
  flex-shrink: 0;
  flex-grow: 0;
  border-right: 1px solid var(--palette-divider);
  margin: 0 calc(2 * var(--spacing-unit));
}
