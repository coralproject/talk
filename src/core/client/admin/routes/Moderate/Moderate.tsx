import key from "keymaster";
import React, {
  FunctionComponent,
  useCallback,
  useEffect,
  useState,
} from "react";

import MainLayout from "coral-admin/components/MainLayout";
import { HOTKEYS } from "coral-admin/constants";
import { PropTypesOf } from "coral-framework/types";
import { SubBar } from "coral-ui/components/v2/SubBar";

import HotkeysModal from "./HotkeysModal";
import ModerateNavigationContainer from "./ModerateNavigation";
import ModerateSearchBarContainer from "./ModerateSearchBar";
import { SiteSelectorContainer } from "./SiteSelector";

import styles from "./Moderate.css";

interface RouteParams {
  storyID?: string;
  siteID?: string;
}

interface Props {
  story: PropTypesOf<typeof ModerateNavigationContainer>["story"] &
    PropTypesOf<typeof ModerateSearchBarContainer>["story"];
  site:
    | { id: string } & PropTypesOf<typeof ModerateNavigationContainer>["site"] &
        PropTypesOf<typeof SiteSelectorContainer>["site"]
    | null;
  query: PropTypesOf<typeof SiteSelectorContainer>["query"];
  moderationQueues: PropTypesOf<
    typeof ModerateNavigationContainer
  >["moderationQueues"];
  allStories: boolean;
  children?: React.ReactNode;
  queueName: string;
  routeParams: RouteParams;
}

const Moderate: FunctionComponent<Props> = ({
  moderationQueues,
  story,
  site,
  query,
  allStories,
  children,
  queueName,
  routeParams,
}) => {
  const [showHotkeysModal, setShowHotkeysModal] = useState(false);
  const closeModal = useCallback(() => {
    setShowHotkeysModal(false);
  }, []);

  useEffect(() => {
    const toggleModal = () => {
      setShowHotkeysModal(was => !was);
    };

    // Attach the modal toggle when the GUIDE button is pressed.
    key(HOTKEYS.GUIDE, toggleModal);
    return () => {
      // Detach the modal toggle if we have to rebind it.
      key.unbind(HOTKEYS.GUIDE);
    };
  }, []);
  return (
    <div data-testid="moderate-container">
      <ModerateSearchBarContainer
        story={story}
        allStories={allStories}
        siteID={routeParams.siteID || null}
        siteSelector={
          <SiteSelectorContainer
            queueName={queueName}
            site={site}
            siteID={routeParams.siteID}
            query={query}
          />
        }
      />
      <SubBar data-testid="moderate-tabBar-container">
        <ModerateNavigationContainer
          moderationQueues={moderationQueues}
          story={story}
          site={site}
        />
      </SubBar>
      <div className={styles.background} />
      <MainLayout data-testid="moderate-main-container">
        <main className={styles.main}>{children}</main>
      </MainLayout>
      <HotkeysModal open={showHotkeysModal} onClose={closeModal} />
    </div>
  );
};

export default Moderate;
