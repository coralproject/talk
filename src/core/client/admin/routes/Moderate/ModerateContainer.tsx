import { Match, RouteProps, Router, withRouter } from "found";
import React from "react";
import { graphql } from "react-relay";

import { withRouteConfig } from "coral-framework/lib/router";
import { Spinner } from "coral-ui/components/v2";

import { ModerateContainerQueryResponse } from "coral-admin/__generated__/ModerateContainerQuery.graphql";

import Moderate from "./Moderate";

interface RouteParams {
  storyID?: string;
}

interface Props {
  data: ModerateContainerQueryResponse;
  router: Router;
  match: Match & { params: RouteParams };
}

class ModerateContainer extends React.Component<Props> {
  public static routeConfig: RouteProps;

  public render() {
    const allStories = !this.props.match.params.storyID;
    if (!this.props.data) {
      return (
        <Moderate moderationQueues={null} story={null} allStories={allStories}>
          <Spinner />
        </Moderate>
      );
    }
    return (
      <Moderate
        moderationQueues={this.props.data.moderationQueues}
        story={this.props.data.story || null}
        allStories={allStories}
      >
        {this.props.children}
      </Moderate>
    );
  }
}

const enhanced = withRouteConfig<Props>({
  query: graphql`
    query ModerateContainerQuery($storyID: ID, $includeStory: Boolean!) {
      # TODO: paginate
      sites {
        edges {
          node {
            id
            ...SiteSelectorSite_site
          }
        }
      }
      story(id: $storyID) @include(if: $includeStory) {
        ...ModerateNavigationContainer_story
        ...ModerateSearchBarContainer_story
      }
      moderationQueues(storyID: $storyID) {
        ...ModerateNavigationContainer_moderationQueues
      }
    }
  `,
  cacheConfig: { force: true },
  prepareVariables: (params, match) => {
    return {
      storyID: match.params.storyID,
      includeStory: Boolean(match.params.storyID),
    };
  },
})(withRouter(ModerateContainer));

export default enhanced;
