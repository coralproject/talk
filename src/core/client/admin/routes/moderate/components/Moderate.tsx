import React, { StatelessComponent } from "react";
import { HorizontalGutter, Typography } from "talk-ui/components";

const Moderate: StatelessComponent = ({ children }) => (
  <HorizontalGutter>
    <Typography variant="heading3">Moderate</Typography>
  </HorizontalGutter>
);

export default Moderate;
