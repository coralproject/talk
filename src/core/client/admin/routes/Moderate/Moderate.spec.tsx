import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { removeFragmentRefs } from "coral-framework/testHelpers";
import { PropTypesOf } from "coral-framework/types";

import Moderate from "./Moderate";

const ModerateN = removeFragmentRefs(Moderate);

it("renders correctly", () => {
  const props: PropTypesOf<typeof ModerateN> = {
    allStories: true,
    moderationQueues: {},
    story: {},
    site: null,
    query: "",
    routeParams: {},
    queueName: "",
  };
  const renderer = createRenderer();
  renderer.render(<ModerateN {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
