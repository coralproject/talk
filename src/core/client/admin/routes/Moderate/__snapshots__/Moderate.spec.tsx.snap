// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="moderate-container"
>
  <ForwardRef(render)
    allStories={true}
    settings={
      Object {
        "multisite": false,
      }
    }
    siteID={null}
    siteSelector={
      <Relay(SiteSelectorContainer)
        query=""
        queueName=""
        site={null}
      />
    }
    story={Object {}}
  />
  <withPropsOnChange(SubBar)
    data-testid="moderate-tabBar-container"
  >
    <Relay(ModerateNavigationContainer)
      moderationQueues={Object {}}
      site={null}
      story={Object {}}
    />
  </withPropsOnChange(SubBar)>
  <div
    className="Moderate-background"
  />
  <MainLayout
    data-testid="moderate-main-container"
  >
    <main
      className="Moderate-main"
    />
  </MainLayout>
  <HotkeysModal
    onClose={[Function]}
    open={false}
  />
</div>
`;
